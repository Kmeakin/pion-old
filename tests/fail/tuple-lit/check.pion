//~ run = "${PION} elab ${FILE}"

{
let _ : (Int, Int, Int, Int) = (false, false, false);
let _ : (Int, Int, Int) = (false, false, false);
let _ : (Int, Int) = (false, false, false);

{}
}
