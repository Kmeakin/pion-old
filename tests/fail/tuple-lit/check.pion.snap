exit status: 0
stdout:
{
    let _ : (Int, Int, Int, Int) = #error;
    let _ : (Int, Int, Int) = #error;
    let _ : (Int, Int) = #error;
    ()
} : ()

stderr:
error: mismatched types
  ┌─ tests/fail/tuple-lit/check.pion:4:32
  │
4 │ let _ : (Int, Int, Int, Int) = (false, false, false);
  │                                ^^^^^^^^^^^^^^^^^^^^^ type mismatch, expected `(Int, Int, Int, Int)`, found `(Bool, Bool, Bool)`

error: mismatched types
  ┌─ tests/fail/tuple-lit/check.pion:5:27
  │
5 │ let _ : (Int, Int, Int) = (false, false, false);
  │                           ^^^^^^^^^^^^^^^^^^^^^ type mismatch, expected `(Int, Int, Int)`, found `(Bool, Bool, Bool)`

error: mismatched types
  ┌─ tests/fail/tuple-lit/check.pion:6:22
  │
6 │ let _ : (Int, Int) = (false, false, false);
  │                      ^^^^^^^^^^^^^^^^^^^^^ type mismatch, expected `(Int, Int)`, found `(Bool, Bool, Bool)`

